### PR TITLE
Add server interface

### PR DESCRIPTION
**Pull request by robertwil**
Extend the bridge by adding a additional node to access the server:

- list server maps
- dowload and save the map as ROS-conform .png + .yaml files for the map_server package

Signed-off-by: Robert Willeit <fkml@gmx.de>

### CODE REVIEW

File: bosch_locator_bridge/CMakeLists.txt [hunk 1]
```diff
@@ -91,9 +93,28 @@ target_link_libraries(${PROJECT_NAME}_node
   Poco::Net
 )
 
+add_executable(${PROJECT_NAME}_server_node
+  src/server/server_main.cpp
+  src/server/server_bridge_node.cpp
+  src/rosmsgs_datagram_converter.cpp
+  src/locator_rpc_interface.cpp)
```
Comment (slaible) [inline bosch_locator_bridge/CMakeLists.txt]:
```suggestion
  src/locator_rpc_interface.cpp
  src/rosmsgs_datagram_converter.cpp
  src/server/server_bridge_node.cpp
  src/server/server_main.cpp)
```
Sort alphabetically

File: bosch_locator_bridge/include/bosch_locator_bridge/server/server_bridge_node.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+class LocatorRPCInterface;
+
+/**
+ * This is the main ROS node. It binds together the ROS interface and the
+ * Locator API.
+ */
+class ServerBridgeNode {
+public:
+  ServerBridgeNode();
+  ~ServerBridgeNode();
```
Comment (slaible) [inline bosch_locator_bridge/include/bosch_locator_bridge/server/server_bridge_node.hpp]:
```suggestion
  virtual ~ServerBridgeNode();
```

**Review by slaible — CHANGES_REQUESTED** [review]
Only tiny changes requested. Otherwise great work!! Thanks a lot :-)

File: bosch_locator_bridge/include/bosch_locator_bridge/server/server_bridge_node.hpp (hunk 2 above)
Comment (slaible) [reply bosch_locator_bridge/include/bosch_locator_bridge/server/server_bridge_node.hpp]: Once merged, I will port it also to ROS 2.

**Comment by robertwil** [comment]
Included your changes (via the browser interface and forgot to sign-off). Hope everything worked as expected, but it looks good.

**Review by slaible — APPROVED** [review]

**Pull request merged**